keylime/registrar_client.py

```diff
@@ -64,7 +68,7 @@ def getData(registrar_ip, registrar_port, agent_id, tls_context):
             logger.critical("Error: did not receive port from Registrar Server.")
             return None
 
-        return response_body["results"]
+        return cast(tenant.RegistrarData, response_body["results"])
```

Comment:
I think that this cast has the same problem. Did you try to write `response_body: tenant.RegistrarData = response.json()`?

But I guess that this would not work either. There is no guarantee that the json is storing a `RegistrarData`.  I think that good approach is to create a function that convert a dict into a RegistrarData, like a sort of type-safe dict.

For example, I found that `inspect` can be used: https://codereview.stackexchange.com/questions/200413/type-safe-json-deserialization

Reply:
I pushed an update. Please have a look.